### PR TITLE
Update SDN/RX references and other minor fixes

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -13,7 +13,7 @@ image:https://img.shields.io/maven-central/v/org.neo4j/neo4j-cypher-dsl.svg[Mave
 
 [abstract]
 --
-The Neo4j Cypher-DSL in its current form is a spin-off from Spring Data Neo4j⚡️RX, where it is used to generate all Cypher queries.
+The Neo4j Cypher-DSL in its current form is a spin-off from Spring Data Neo4j⚡️RX (now Spring Data Neo4j 6+), where it is used to generate all Cypher queries.
```
Comment (michael-simons) [inline README.adoc:16]:
```suggestion
The Neo4j Cypher-DSL in its current form is a spin-off from Spring Data Neo4j 6+ (nè Spring Data Neo4j⚡️RX), where it is used to generate all Cypher queries.
```

File: docs/index.adoc [hunk 2]
```diff
@@ -39,7 +39,7 @@ This manual is written for people interested in creating Cypher queries in a typ
 WARNING: The Cypher-DSL is considered _EXPERIMENTAL_ at the time of writing.
 This means you can and should use it and that we are looking actively for feedback, issues and problems.
 We will try to keep breaking changes to a minimum, but we reserve the right to modify public methods, parameters
-and behaviour in case we need those changes for the https://github.com/neo4j/sdn-rx[SDN/RX] project, the origin of the Cypher-DSL.
+and behaviour in case we need those changes for the https://github.com/spring-projects/spring-data-neo4j[SDN] project.
```
Comment (michael-simons) [inline docs/index.adoc:42]: I would rather drop the referencers in the text  all together.

File: docs/introduction-and-preface/introduction.adoc [hunk 3]
```diff
@@ -1,12 +1,12 @@
 == Purpose
 
-The Cypher-DSL has been developed with the needs of https://github.com/neo4j/sdn-rx[SDN/RX] in mind:
+The Cypher-DSL has been developed with the needs of https://github.com/neo4j/sdn-rx[SDN/RX] in mind (now superseded by https://github.com/spring-projects/spring-data-neo4j[SDN]):
```
Comment (michael-simons) [inline docs/introduction-and-preface/introduction.adoc:3]: Same as above

**Review by michael-simons — CHANGES_REQUESTED** [review]
Thank you. Great PR. Just minor things to change, please.

File: README.adoc (hunk 1 above)
Comment (fbiville) [reply README.adoc:16]: that would be `né` actually (unless there is a typo like in `connoisseur` instead of `connaisseur`) :)

**Pull request merged**